### PR TITLE
chore: make a single switcherapi class

### LINKED ISSUE #778: Refactor API Instantiation
### What did you have in mind?

https://github.com/TomerFi/aioswitcher/pull/777#discussion_r1674562168

### Are you trying to fix a problem?

_No response_

### Any lead on how this feature can be implemented?

_No response_

### PR DESCRIPTION
**Pull request by YogevBokobza**
## Description

BREAKING CHANGE: Convert SwitcherType1Api and SwitcherType2Api to a single SwitcherApi class.

**Related issue (if any):** fixes #778 

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I will adhere to the project's code of conduct.

### CODE REVIEW

**Comment by YogevBokobza** [comment]
Does adding "IF"s to each function make sense to allow the function to run only for a specific device type?
I mean this function "stop_shutter(..)" is meant for Type2 devices and not Type1. Should we block accessing Type1 devices for this function?

File: src/aioswitcher/api/__init__.py [hunk 1]
```diff
@@ -101,15 +101,17 @@ def __init__(
...
         """Initialize the Switcher TCP connection API."""
         self._device_type = device_type
         self._ip_address = ip_address
         self._device_id = device_id
         self._device_key = device_key
-        self._port = port
+        if self._device_type.protocol_type == 1:
+            self._port = SWITCHER_TCP_PORT_TYPE1
+        else:
+            self._port = SWITCHER_TCP_PORT_TYPE2
```
Comment (thecode) [inline src/aioswitcher/api/__init__.py]: We have a mapping that is now not in use anywhere in the project:
https://github.com/TomerFi/aioswitcher/blob/b72c9de19c7a87ac260dd34f8350caaa0ff019f4/src/aioswitcher/api/__init__.py#L66-L74
Make sure to remove it

As for this, I usually prefer to make it shorter like this:
```suggestion
        self._port = SWITCHER_TCP_PORT_TYPE1
        if device_type.protocol_type == 2:
            self._port = SWITCHER_TCP_PORT_TYPE2
```

Comment (YogevBokobza) [reply src/aioswitcher/api/__init__.py]: Oh I didn't notice the mapping..
Should I just use the mapping? or keep what I did and remove the mapping?

Comment (thecode) [reply src/aioswitcher/api/__init__.py]: Remove the mapping

Comment (YogevBokobza) [reply src/aioswitcher/api/__init__.py]: It is done.

**Comment by thecode** [comment]
> Does adding "IF"s to each function make sense to allow the function to run only for a specific device type? I mean this function "stop_shutter(..)" is meant for Type2 devices and not Type1. Should we block accessing Type1 devices for this function?

I don't think we should complicate the implementation with this

**Comment by YogevBokobza** [comment]
> > Does adding "IF"s to each function make sense to allow the function to run only for a specific device type? I mean this function "stop_shutter(..)" is meant for Type2 devices and not Type1. Should we block accessing Type1 devices for this function?
> 
> I don't think we should complicate the implementation with this

Fine by me.

**Review by thecode — APPROVED** [review]

**Review by TomerFi — APPROVED** [review]

**Comment by TomerFi** [comment]
Thank you very much!
Merged and released: [5.0.0](https://github.com/TomerFi/aioswitcher/releases/tag/5.0.0).

**Comment by TomerFi** [comment]
@YogevBokobza, @thecode, I wonder if anyone uses the bridge with anything other than the default four ports. Do we even need this machinery?

https://github.com/TomerFi/aioswitcher/blob/dev/src/aioswitcher/bridge.py#L382

**Comment by thecode** [comment]
> @YogevBokobza, @thecode, I wonder if anyone uses the bridge with anything other than the default four ports. Do we even need this machinery?
> 
> https://github.com/TomerFi/aioswitcher/blob/dev/src/aioswitcher/bridge.py#L382

No, it is not possible to use any other port anyhow, I think I added it at first because I thought about running more than 1 bridge with different ports, I guess this can be removed, while going over this, I also noticed we have the following mapping
https://github.com/TomerFi/aioswitcher/blob/0027863d7ea0f01ee0be786496ac8f1039fd8277/src/aioswitcher/bridge.py#L64C1-L72
which I am not sure where and if it is used

**Comment by TomerFi** [comment]
> I also noticed we have the following mapping
https://github.com/TomerFi/aioswitcher/blob/0027863d7ea0f01ee0be786496ac8f1039fd8277/src/aioswitcher/bridge.py#L64C1-L72

Agreed, this can be removed. 
We'll open another issue for more refactoring.

**Pull request merged**